src/components/dropdown-search/dropdown-search.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import "./dropdown-search.scss";
+import ArrowDown from "../../images/icons/arrow-down.svg";
+import { SearchInput } from "../..";
+
+export const DropdownSearch = ({
+  title,
+  values,
+  onSelect,
+  isAlwaysOpen,
+  showSearchInput = true,
+  searchPlaceholder
+}) => {
+  const [showDropdownOptions, setShowDropdownOptions] = useState(false);
+  const [dropdownTitle, setDropdownTitle] = useState(title);
+  const [dropdownOptions, setDropDownOptions] = useState(values);
+  const [searchInput, setSearchInput] = useState("");
+
+  const selectDropdownOption = option => {
+    setDropdownTitle(option);
+    setShowDropdownOptions(!showDropdownOptions);
+    setSearchInput("");
+    setDropDownOptions(values);
+    onSelect(option);
+  };
+
+  const toggleDropdownOptions = () => {
+    setShowDropdownOptions(!showDropdownOptions);
+  };
+
+  const filterCaseInsensitiveElementsContaining = text => {
+    if (text && text.trim() === "") {
+      return values;
+    }
+    return values.filter(element =>
+      element.toLowerCase().includes(text.toLowerCase())
+    );
+  };
+
+  const filterValues = searchText => {
+    setSearchInput(searchText);
+    setDropDownOptions(filterCaseInsensitiveElementsContaining(searchText));
+  };
+
+  return (
+    <div>
+      <div className="dropdown-search-filter" onClick={toggleDropdownOptions}>
+        <div className="dropdown-search-filter__name">{dropdownTitle}</div>
+        <span className="icon is-small is-right dropdown-search-filter__icon">
+          <ArrowDown />
+        </span>
+      </div>
+      {(showDropdownOptions || isAlwaysOpen) && (
+        <div className={"dropdown-search-options"}>
+          {showSearchInput && (
+            <>
+              <SearchInput
+                value={searchInput}
+                onValueChange={filterValues}
+                placeholder={searchPlaceholder}
+              />
+              <hr />
+            </>
+          )}
+          <div className="dropdown-search-options__scrollable-wrapper">
+            <div className="dropdown-search-options__list">
+              {dropdownOptions.map((option, index) => {
+                return (
+                  <div
+                    className={"dropdown-search-options__value"}
+                    key={index}
+                    id={option}
+                    onClick={() => {
+                      selectDropdownOption(option);
+                    }}
+                  >
+                    {option}
+                  </div>
+                );
+              })}
+            </div>
+          </div>
+        </div>
+      )}
+    </div>
+  );
+};
+
+DropdownSearch.defaults = {
+  title: "",
+  values: [],
+  onSelect: () => {},
+  isAlwaysOpen: false,
+  showSearchInput: true,
+  searchPlaceholder: ""
+};
+
+DropdownSearch.propTypes = {
+  title: PropTypes.string.isRequired,
+  values: PropTypes.array.isRequired,
```

Comment:
I would rename this prop `options` and would expect 
```
PropTypes.arrayOf(
  PropTypes.shape({
    value:PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number
    ]),
    label: PropTypes.string
  })
)
```
as the values from the database could be different than what user should see. What do you think?

Reply:
this is how the model on backend will look like: https://github.com/code4romania/stam-acasa/pull/169/files#diff-10e2322e3b329f8be005addfd05e4368, we want to keep the ids in `value` key and the names in `label` key

Reply:
That's cool, and just to clarify, the returned value to onSelect should be the ID, and not the label, right? Or do we want both?

Reply:
`selectDropdownOption(option.value)` should be enough